Soviet Soldier Volkov & Chitzkoi fully ported

AngryBirdz opened the pull request:
Hi there, I've just completed my porting of the Counterstrike expansion mission named "Soviet Soldier Volkov & Chitzkoi" (scu35ea). The map is fully operational but it might need some playtest by someone else than myself just in case there's still some bugs and / or balance issue that I did'nt noticed.

Have a nice day :)

Commit:
Soviet Soldier Volkov & Chitzkoi fully ported

Crash fixed.

reaperrr commented:
It looks like our lua syntax checker can't handle the '&' in the folder name, maybe just write 'and' instead.

It would also be nice if you could list the lua scripts in https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.sln, some reviewers might use that for debugging.

pchote commented:
The `&` character has special meaning (put job in background) on Linux and macOS, and so can't be used unescaped in file paths.  Yes, please replace it with `and` or something else.

AngryBirdz commented:
Alright guys I did everything you asked and I think I did'nt forget anything.

UPDATE
-Added myself in the authors file (I forgot the first time)
-Removed the special character in the filename
-Listed my lua scripts in OpenRA.sln
-Removed two useless line in weapons.yaml (just a leftover of me fooling around with some stuff)

AngryBirdz commented:
Can you explain to me why this has happened? I seriously don't know. I double checked my code and the comp "Goodguy" is listed both in the map.yaml and the lua script. What did I miss?

pchote commented:
Player names are case sensitive.  See `GoodGuy` vs `Goodguy`.

pchote commented on AUTHORS:
```diff
@@ -104,6 +104,7 @@ Also thanks to:
     * Michael Rätzel
     * Michael Silber (frühstück)
     * Michael Sztolcman (s1w_)
+	* Mike Gagné (AngryBirdz)
```
Please use 4 spaces here instead of a tab.

pchote commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/rules.yaml:
```diff
@@ -0,0 +1,233 @@
+		Prerequisites: ~syrd, ~techlevel.high
+        
+C2:
+	Inherits@AUTOTARGET: ^AutoTargetGroundAssaultMove
```
Have you tried defining `Inherits@2: ^ArmedCivilian` instead?  This would save a lot of duplication.

AngryBirdz replied:
Yeah I'll upload the new file soon

pchote commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/rules.yaml:
```diff
@@ -0,0 +1,233 @@
+		Range: 10c0
+	Demolition:
+	Mobile:
+		-Voice: Move
```
Why do these voices need to be removed?

AngryBirdz replied:
Because these voices can't be used with the GenericVoice voiceset. It make the game crash.

Mailaender commented:
A very minor issue:

![image](https://user-images.githubusercontent.com/756669/45289582-d0e94f00-b4ed-11e8-9917-93d2b556facb.png)

The mines display the multiplayer user name on hover.

Mailaender commented:
Another polishing issue:

![image](https://user-images.githubusercontent.com/756669/45290086-fb87d780-b4ee-11e8-87a1-252eb9b761bd.png)

The lab (and hospital) still shows it's multiplayer function in the tooltip.

Mailaender commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea-AI.lua:
```diff
@@ -0,0 +1,120 @@
+  local Rally = Utils.Random(AlliedWarFactRally)
+  Utils.Do(AlliedWarFact, function(fact)
+    fact.RallyPoint = Rally.Location
+  end)
```
Indention look wrong here.

Mailaender commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea-AI.lua:
```diff
@@ -0,0 +1,120 @@
+	elseif (OreTruck01.IsDead and OreTruck02.IsDead) and greece.Resources <= 2399 then
+		return
+	end
+  NavalYard01.RallyPoint = waypoint26.Location
```
You seem to be using a mix of tabs and spaces for indention. Please just use tabs.

Mailaender reviewed (dismissed):
Looking good otherwise. Please squash all fixup commits together when you are done.

AngryBirdz commented:
I'm a noob at this. Can you explain to me what this "PR: Needs +2" label mean? Did I forgot to do something?

Smittytron commented:
OpenRA requires two approval reviews to merge a PR. The tag just means there's one approval.

AngryBirdz commented:
Ok thx for the info.

abcdefg30 commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea.lua:
```diff
@@ -0,0 +1,380 @@
+		ProduceInfantry() --Greece will start infantry production once this unit is dead just like in the original mission
+	end)
+
+--Kill the ore trucks when the two allied refinery are destroyed
```
Why is this necessary? (It looks a bit odd to me.)

AngryBirdz replied:
The vanilla mission work that way.

abcdefg30 replied:
Can you please remove it (still)? I would consider that an improvement over the original mission. (It does not make sense to me why the ore trucks should blow up. As player, I would probably be rather confused by this.)

abcdefg30 reviewed (commented):
The tooltip of the alloy factory needs to be hidden/adjusted:
![grafik](https://user-images.githubusercontent.com/7704140/45559921-0a9abc80-b844-11e8-9b82-5c134c7777dc.png)

Otherwise, well done! The mission plays nicely ingame. Only issue I had: The allied forces often only damage the concrete walls of your base and you still get a "Our base is under attack" warning which I found to be  a bit annoying/distracting (for just concrete...). Do you think this should be changed?

abcdefg30 commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea.lua:
```diff
@@ -0,0 +1,380 @@
+		Media.PlaySpeechNotification(player, "ReinforcementsArrived")
+		transport.Paradrop(CPos.New(21, 82))
+	end)
+ 
```
I forgot to mention: trailing white space. :)

AngryBirdz commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/rules.yaml:
```diff
@@ -0,0 +1,207 @@
+		DefaultCash: 0
+	LobbyPrerequisiteCheckbox@GLOBALBOUNTY:
+		Enabled: False
+		Locked: True
```
Are you sure that LobbyPrerequisiteCheckbox@GLOBALBOUNTY is pre-defined in campaign-rules.yaml? Because I don't see it.

pchote replied:
@abcdefg30 Do we need it? Bounties default to off in the default game rules.

abcdefg30 replied:
No, we don't necessarily need this. However, we usually lock the options in the campaign missions.

AngryBirdz commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/map.yaml:
```diff
@@ -0,0 +1,1449 @@
+		Name: Creeps
+		NonCombatant: True
+		Faction: england
+        Enemies: Greece, GoodGuy, Spain, France, USSR
```
For the tooltip description i have two questions:
1) How do you make it show ingame? Can't find the setting.
2) Do you know wich trait point to it? Because I'm having a hard time finding it.
Thx!

abcdefg30 replied:
The tooltip should just appear when you hover with your mouse over the building. The trait is `TooltipDescription` iirc.

AngryBirdz commented:
I tried to fix the Tooltip Description with the trait you told me to mod but the check failed somehow?

Mailaender commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/rules.yaml:
```diff
@@ -0,0 +1,204 @@
+		GenericName: Hospital
+		GenericVisibility: Enemy, Ally, Neutral
+		GenericStancePrefix: false
+		TooltipDescription: Destroy this hospital to get healing equipment.
```
This is indented wrong. `TooltipDescription` is it's own trait:

```
	TooltipDescription:
		Description: Destroy this hospital to get healing equipment.
```

should work.

AngryBirdz replied:
![notworking](https://user-images.githubusercontent.com/42563347/45836068-d0b13680-bcd9-11e8-8051-8a684ba735c8.PNG)
Not sure if by its own trait you mean outside of tooltip but if I'm doing it like that the game wont even load the map.

![working](https://user-images.githubusercontent.com/42563347/45836197-24238480-bcda-11e8-943e-07387bf275e6.PNG)
On the other hand if I indent it like you told me but inside tooltip the map load. Can you confirm this will work before I update the commit? Thanks you.

pchote replied:
[TooltipDescription](https://github.com/OpenRA/OpenRA/blob/d33ac0c8384ae360b725b5886f4c28ade95af315/OpenRA.Mods.Common/Traits/TooltipDescription.cs) is a different trait to [Tooltip](https://github.com/OpenRA/OpenRA/blob/d33ac0c8384ae360b725b5886f4c28ade95af315/OpenRA.Mods.Common/Traits/Tooltip.cs).  When you indent the definition as in your second example the game tries to load a `TooltipDescription` *field* on the `Tooltip` trait, which does not exist and is therefore ignored.

The lint check (which you can view from the "Details" link in the "Some checks were not successful" panel below reports this more explicitly:
```
OpenRA.Utility(1,1): Error: rules.yaml:75 refers to a trait field `TooltipDescription` that does not exist on `Tooltip`.
OpenRA.Utility(1,1): Error: rules.yaml:101 refers to a trait field `TooltipDescription` that does not exist on `Tooltip`.
```

Mailaender replied:
> Not sure if by its own trait you mean outside of tooltip but if I'm doing it like that the game wont even load the map.

How do you test this map? Your local OpenRA installation or this branch compiled from source?

AngryBirdz replied:
My local OpenRA installation.

Mailaender commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/rules.yaml:
```diff
@@ -0,0 +1,204 @@
+		GenericName: Alloy Facility
+		GenericVisibility: Enemy, Ally, Neutral
+		GenericStancePrefix: false
+		TooltipDescription: This facility produce our stolen armor plating.
```
Same here.

abcdefg30 replied:
While you are at it, please change `produce` to `produces`.

AngryBirdz commented:
I've managed to fix it by using the playtest build. Just tell me if I forgot something.

abcdefg30 commented on mods/ra/maps/soviet-soldier-volkov-n-chitzkoi/scu35ea.lua:
```diff
@@ -0,0 +1,376 @@
+				end
+			end)
+			if Map.LobbyOption("difficulty") == "hard" then
+				Trigger.ClearAll(RiflemanGuard01)
```
This crashes when `RiflemanGuard01` is already dead.

AngryBirdz commented:
Ok I've fixed the crash related to RiflemanGuard01. Tell me if there's something else.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b8a3afc425a2588092f098e49b9e0f251723972e...0c6096c3c6d91c95ade6036751abe3481d52e9c5)
